app/common/model/src/main/java/io/syndesis/common/model/integration/step/template/AbstractTemplatePreProcessor.java

```diff
@@ -231,7 +236,7 @@ public void reset() {
 
     @Override
     public Map<String, Object> getUriParams() {
-        return Collections.emptyMap();
+        return new HashMap<>(HEADER_TEMPLATE_OPTIONS);
```

Comment:
Wonder if we can change the constructor of AbstractTemplatePreProcessor to accept a map, so specific implementation can provide their own uri params when the instance is created ora have a protected `Map<String, Object> uriParams` that is amended by the subclasses so then we can have something like:

```java
public Map<String, Object> getUriParams() {
    return Collections.unmodifiableMap(uriParams);
}
```

Reply:
I think the problem is caused by MustacheTemplatePreProcessor. I'm bringing back the abstract method to return an final map and then do the same on the Mustache subclass, so, all returned maps will be final.